plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/BaseSqlServerTypeMapping.java

```diff
@@ -430,67 +437,44 @@ public void testVarbinary()
+    {
+        // BC dates not supported by SQL Server
+        return SqlDataTypeTest.create()
                 .addRoundTrip("date", "NULL", DATE, "CAST(NULL AS DATE)")
                 // first day of AD
-                .addRoundTrip("date", "DATE '0001-01-01'", DATE, "DATE '0001-01-01'")
-                .addRoundTrip("date", "DATE '0012-12-12'", DATE, "DATE '0012-12-12'")
+                .addRoundTrip("date", inputLiteralFactory.apply("'0001-01-01'"), DATE, "DATE '0001-01-01'")
+                .addRoundTrip("date", inputLiteralFactory.apply("'0012-12-12'"), DATE, "DATE '0012-12-12'")
                 // before julian->gregorian switch
-                .addRoundTrip("date", "DATE '1500-01-01'", DATE, "DATE '1500-01-01'")
+                .addRoundTrip("date", inputLiteralFactory.apply("'1500-01-01'"), DATE, "DATE '1500-01-01'")
                 // before epoch
-                .addRoundTrip("date", "DATE '1952-04-03'", DATE, "DATE '1952-04-03'")
-                .addRoundTrip("date", "DATE '1970-01-01'", DATE, "DATE '1970-01-01'")
-                .addRoundTrip("date", "DATE '1970-02-03'", DATE, "DATE '1970-02-03'")
+                .addRoundTrip("date", inputLiteralFactory.apply("'1952-04-03'"), DATE, "DATE '1952-04-03'")
+                .addRoundTrip("date", inputLiteralFactory.apply("'1970-01-01'"), DATE, "DATE '1970-01-01'")
+                .addRoundTrip("date", inputLiteralFactory.apply("'1970-02-03'"), DATE, "DATE '1970-02-03'")
                 // summer on northern hemisphere (possible DST)
-                .addRoundTrip("date", "DATE '2017-07-01'", DATE, "DATE '2017-07-01'")
+                .addRoundTrip("date", inputLiteralFactory.apply("'2017-07-01'"), DATE, "DATE '2017-07-01'")
                 // winter on northern hemisphere (possible DST on southern hemisphere)
-                .addRoundTrip("date", "DATE '2017-01-01'", DATE, "DATE '2017-01-01'")
-                .addRoundTrip("date", "DATE '1970-01-01'", DATE, "DATE '1970-01-01'")
-                .addRoundTrip("date", "DATE '1983-04-01'", DATE, "DATE '1983-04-01'")
-                .addRoundTrip("date", "DATE '1983-10-01'", DATE, "DATE '1983-10-01'");
-
-        for (String timeZoneId : ImmutableList.of(UTC_KEY.getId(), jvmZone.getId(), someZone.getId())) {
-            Session session = Session.builder(getSession())
-                    .setTimeZoneKey(TimeZoneKey.getTimeZoneKey(timeZoneId))
-                    .build();
-            testsSqlServer.execute(getQueryRunner(), session, sqlServerCreateAndInsert("test_date"));
-            testsTrino.execute(getQueryRunner(), session, trinoCreateAsSelect(session, "test_date"));
-            testsTrino.execute(getQueryRunner(), session, trinoCreateAndInsert(session, "test_date"));
-        }
+                .addRoundTrip("date", inputLiteralFactory.apply("'2017-01-01'"), DATE, "DATE '2017-01-01'")
+                .addRoundTrip("date", inputLiteralFactory.apply("'1970-01-01'"), DATE, "DATE '1970-01-01'")
```

Comment:
Could you remove this line because the case `1970-01-01` already exists in L469? I know it's pre-existing issue though.

Reply:
I think we should remove L469 and keep L475 (you can refer to [TestPostgreSqlTypeMapping.java#L1092](https://github.com/trinodb/trino/blob/master/plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestPostgreSqlTypeMapping.java#L1092) to see the purpose of the L475)

Besides, some other date type tests have the same issue (e.g. BigQuery, ClickHouse, PostgreSQL)

And I also think we should uniform the style of date type tests (e.g. comments are put at the end of the line), I think `TestPostgreSqlTypeMapping#testDate` is a good example.

WDYT @ebyhr

Reply:
> I think we should remove L469 and keep L475

I don't think so. It would be better to order the dates (old → new) so that we can understand what date is missing quickly.

Reply:
Got it, thanks for the explanation.

Reply:
> Could you remove this line because the case 1970-01-01 already exists in L469?

done